hspec-core/test/Test/Hspec/Core/FormattersSpec.hs

```diff
@@ -280,6 +281,14 @@ spec = do
     context "same as failed_examples" $ do
       failed_examplesSpec formatter
 
+  describe "additional formatter features" $ do
+    describe "getFinalCount" $ do
+      let formatter = H.silent {footerFormatter = fmap show getFinalCount >>= writeLine}
+          runSpec = captureLines . H.hspecWithResult H.defaultConfig {H.configFormatter = Just formatter}
+      it "counts examples" $ do
+        result:_ <- runSpec testSpec
+        result `shouldBe` "6"
```

Comment:
Actually, I think we are indirectly testing `countExamplesInSpec` here.  Remove this and add a test for `countExamplesInSpec` instead?

Reply:
You're correct, and I can certainly add a unit test for `countSpecItems` if you like, but:
1. I think having a simple end-to-end test ensuring the API is wired up correctly does add value, and
2. If this test exists, given how simple `countSpecItems` is, I don't know if a unit test adds much additional value. If anything, I would spend my next testing budget dollar on adding more end-to-end tests with more complex specs (with hooks, with more/fewer `describe` sections, etc.).

What do you think?

Reply:
> I think having a simple end-to-end test ensuring the API is wired up correctly does add value

I'm ok with that.  So lets keep it as is for now.